### PR TITLE
Update Windows Files

### PR DESCRIPTION
**Pull request by HarrisonWAffel**
This PR closes the gap between how files are created on Linux and Windows. It also adds unit tests around file permissions, as well as updates CI to run these tests on both Linux and Windows. 

Note: This unit test does not compile properly on arm, and I can't determine why. Testing the CI on my fork repeatedly resulted in the below error. Because of this, I added a build tag to exclude this unit test on arm. I'm not aware of any differences between file creation and permissions between the two architectures, but if we feel strongly about including this test on arm I can continue to look into the error. But ultimately, for linux, this unit test is covering functionality was added to this repo ~3 years ago.

```  
fork/exec /tmp/go-build3394345592/b1239/applyinator.test: exec format error
FAIL	github.com/rancher/system-agent/pkg/applyinator	0.002s
```

### CODE REVIEW

**Review by jakefhyde — DISMISSED** [review]
Approved, weird about the arm test

File: pkg/applyinator/file_test.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+		{
+			Path:    "test-no-perms",
+			Content: "hello world",
+
+			ExpectedPermissions: defaultFilePermissions,
+			ExpectedErr:         false,
+		},
+		{
+			Path:        "test-perms",
+			Permissions: "0600",
```
Comment (jiaqiluo) [inline pkg/applyinator/file_test.go:49]: it seems that `defaultFilePermissions` is also 0600, could we change the value here, and in other cases, to a different one?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
LGTM overall, with only one comment.

File: pkg/applyinator/file_test.go (hunk 1 above)
Comment (HarrisonWAffel) [reply pkg/applyinator/file_test.go]: I've updated the perms in `TestWriteContentToFile` to `0666`

**Review by jiaqiluo — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**